### DISCUSSION TITLE
How to delete input field name with value before submitted form

### DISCUSSION THREAD

**Discussion by abdulmateentechbits** [react-hook-form/react-hook-form#9220] [Q&A] [answered]
Hi kindly help me  
I want when user chose percent delete price field name and value and if he chose fixed amount delete the percent field with value before submitted the form
like this
"locations": [
{
"location": "Main Boulevard",
"is_fixed": "1", //Amount selected percent absent
"price": "100000"
},
{
"location": "Corner",
"is_fixed": "0", // Percent selected price is absent
"percent": "5"
}
My code sandbox
https://codesandbox.io/s/react-hook-form-forked-rdesqv?file=/src/components/forms/basics/DynamicFields.jsx

@bluebill1049
@leapful
@evocateur
@aaronjensen

**Comment by bluebill1049** [comment, marked as answer]
could you do that on the onSubmit callback? just omit those value which you don't wan to send to the server.

**Reply by Moshyfawn** [reply]
Yea, it's best to run your field exclusion logic inside `onSubmit`. This way you don't need to watch field values.

Now, if you wanna hide inputs based on your user selection and omit the field values all together, you can unmount those inputs with JS _(conditions?)_ and set `useForm` prop [`shouldUnregister: true`](https://react-hook-form.com/api/useform).
